bitnami/concourse/templates/worker/deployment.yaml

```diff
@@ -0,0 +1,217 @@
+            {{- end }}
+            {{- if .Values.worker.baggageclaim.btrfsBin }}
+            - name: CONCOURSE_BAGGAGECLAIM_BTRFS_BIN
+              value: {{ .Values.worker.baggageclaim.btrfsBin | quote }}
+            {{- end }}
+            {{- if .Values.worker.baggageclaim.mkfsBin }}
+            - name: CONCOURSE_BAGGAGECLAIM_MKFS_BIN
+              value: {{ .Values.worker.baggageclaim.mkfsBin | quote }}
+            {{- end }}
+            {{- if .Values.worker.baggageclaim.overlaysDir }}
+            - name: CONCOURSE_BAGGAGECLAIM_OVERLAYS_DIR
+              value: {{ .Values.worker.baggageclaim.overlaysDir | quote }}
+            {{- end }}
+            {{- if .Values.worker.baggageclaim.disableUserNamespaces }}
+            - name: CONCOURSE_BAGGAGECLAIM_DISABLE_USER_NAMESPACES
+              value: {{ .Values.worker.baggageclaim.disableUserNamespaces | quote }}
+            {{- end }}
+          envFrom:
+            {{- if .Values.worker.extraEnvVarsCM }}
+            - configMapRef:
+                name: {{ include "common.tplvalues.render" (dict "value" .Values.worker.extraEnvVarsCM "context" $) }}
+            {{- end }}
+            {{- if .Values.worker.extraEnvVarsSecret }}
+            - secretRef:
+                name: {{ include "common.tplvalues.render" (dict "value" .Values.worker.extraEnvVarsSecret "context" $) }}
+            {{- end }}
+          {{- if .Values.worker.resources }}
+          resources: {{- toYaml .Values.worker.resources | nindent 12 }}
+          {{- end }}
+          {{- if .Values.worker.livenessProbe.enabled }}
+          livenessProbe: {{- include "common.tplvalues.render" (dict "value" (omit .Values.worker.livenessProbe "enabled") "context" $) | nindent 12 }}
+          {{- else if .Values.worker.customLivenessProbe }}
+          livenessProbe: {{- include "common.tplvalues.render" (dict "value" .Values.worker.customLivenessProbe "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.worker.readinessProbe.enabled }}
+          readinessProbe: {{- include "common.tplvalues.render" (dict "value" (omit .Values.worker.readinessProbe "enabled") "context" $) | nindent 12 }}
+          {{- else if .Values.worker.customReadinessProbe }}
+          readinessProbe: {{- include "common.tplvalues.render" (dict "value" .Values.worker.customReadinessProbe "context" $) | nindent 12 }}
+          {{- end }}
+          volumeMounts:
+            - name: concourse-keys
+              mountPath: {{ .Values.worker.keySecretsPath | quote }}
+              readOnly: true
+          {{- if .Values.worker.extraVolumeMounts }}
+          {{- include "common.tplvalues.render" (dict "value" .Values.worker.extraVolumeMounts "context" $) | nindent 12 }}
+          {{- end }}
+        {{- if .Values.worker.sidecars }}
+        {{- include "common.tplvalues.render" ( dict "value" .Values.worker.sidecars "context" $) | nindent 8 }}
+        {{- end }}
+      volumes:
+        - name: concourse-keys
```

Comment:
In the statefulset version we have a volume for the work dir. In the case of deployment (and only in the case of deployment) we should have a pvc.yaml that would be enabled if worker.persistence.enabled and worker.deploymentType = 'deployment'. In the case of statefulsets we already have the volumeClaimTemplote

Reply:
Still not done

Reply:
The deployment doesn't have the work dir, only have the concourse-key and is a secret

Reply:
But if it doesn't havet the workdir, it will not work the same as with the statefulset, it would have less value using this approach than statefulsets, right?